packages/dds/tree/src/feature-libraries/chunked-forest/uniformChunk.ts

```diff
@@ -93,11 +98,29 @@ export class TreeShape {
 	// TODO: this is only needed at chunk roots. Optimize it base on that.
 	public readonly positions: readonly NodePositionInfo[];
 
+	/**
+	 *
+	 * @param type - {@link TreeNodeSchemaIdentifier} used to compare shapes.
+	 * @param hasValue - whether or not the TreeShape has a value.
+	 * @param fieldsArray - an array of {@link FieldShape} values, which contains a TreeShape for each FieldKey.
+	 *
+	 * @param maybeDecompressedStringAsNumber - used to check whether or not the value could have been compressed by the idCompressor.
+	 * This flag can only be set on string leaf nodes, and will throw a usage error otherwise.
+	 * If set to true, an additional check can be made (example: getting the value of {@link Cursor}) to return the original uncompressed value.
+	 */
 	public constructor(
 		public readonly type: TreeNodeSchemaIdentifier,
 		public readonly hasValue: boolean,
 		public readonly fieldsArray: readonly FieldShape[],
+		public readonly maybeDecompressedStringAsNumber: boolean = false,
 	) {
+		if (maybeDecompressedStringAsNumber) {
```

Comment:
nit: why not have both checks in one if statement?

Reply:
combined both checks into single if statement